src/views/Players/PlayerDetails.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import { useHistory, useParams } from 'react-router'
+import { useEffect, useState } from 'react/cjs/react.development';
+import PlayerCard from '../../components/PlayerCard/PlayerCard.jsx';
+import PlayerForm from '../../components/PlayerForm/PlayerForm.jsx';
+import { deletePlayerById, getPlayerById, updatePlayerById } from '../../services/players.js';
+
+export default function PlayerDetails() {
+    const { id } = useParams();
+
+    const history = useHistory();
+
+    const [playerNameStr, setPlayerNameStr] = useState('');
+    const [playerPositionStr, setPlayerPositionStr] = useState('');
+    const [playerDataObj, setPlayerDataObj] = useState({});
+    const [isLoading, setIsLoading] = useState(true);
+    const [editSwitch, setEditSwitch] = useState(false);
+
+
+    // ❓ Is there a better way to approach this?
+    async function fetchPlayerData(id) {
+        setIsLoading(true);
+        const newPlayerDataObj = await getPlayerById(id);
+        setPlayerDataObj(newPlayerDataObj);
+        setPlayerNameStr(newPlayerDataObj.name);
+        setPlayerPositionStr(newPlayerDataObj.position)
+        setIsLoading(false);
+    }
+
+
+    // ❓ Is there a better way to approach this? I had to add the id as a param to fetchTeamData fn and then pass it as a dependency. If I feed it the fetchTeamData fn as a dependency, it creates an infinite loop. If I remove the array, it creates an infinite loop.
```

Comment:
this is the correct approach!
`no dependency array`: run on mount and _every re-render_ (aka any time any state variable changes)
`variable(s) in the dependency array`: run on mount and _only when this variable changes_
`empty dependency array`: run on mount only (but eslint yells at you to put the variable in the dependency array)